### ISSUE TITLE
Incomplete ISO download makes startup fail

### ISSUE THREAD

**Issue by cmoulliard** [minishift/minishift#344]
Version used : 1.0.0.Beta3

error 

```
minishift start --openshift-version=v1.4.1 --memory=4000 --vm-driver=virtualbox --iso-url=https://github.com/minishift/minishift-centos-iso/releases/download/v1.0.0-rc.1/minishift-centos7.iso --docker-env=[storage-driver=devicemapper]
Starting local OpenShift cluster using 'virtualbox' hypervisor...
Downloading ISO 'https://github.com/minishift/minishift-centos-iso/releases/download/v1.0.0-rc.1/minishift-centos7.iso'
 295.52 MB / 329.00 MB [==================================================================================================================================================================>------------------]  89.82% 13s
E0130 11:01:36.831166    3152 start.go:135] Error starting the VM: read tcp 192.168.1.80:53769->52.216.64.240:443: read: operation timed out. Retrying.
^C
dabou-macosx:~$ minishift start --openshift-version=v1.4.1 --memory=4000 --vm-driver=virtualbox --iso-url=https://github.com/minishift/minishift-centos-iso/releases/download/v1.0.0-rc.1/minishift-centos7.iso --docker-env=[storage-driver=devicemapper]
Starting local OpenShift cluster using 'virtualbox' hypervisor...
E0130 11:16:50.500950    3660 start.go:135] Error starting the VM: Error starting stopped host: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded. Retrying.
Password:
```

**Comment by hferentschik** [comment]
ok, maybe a comment from issue #343 was actually supposed to go on this issue. I think we have a bug here.

It looks like the following happened. The ISO downloaded started and got aborted. My guess is the file was partially downloaded and existed on the file system. Our current checks only verify the ISO based on file name, so on the second attempt, we assume there is a valid ISO, but it is actually corrupted. See also https://github.com/minishift/minishift/blob/9d320e02683c857aa395e2adcc65fcf42a68db64/pkg/minikube/cluster/cluster.go#L261-L261

So our algorithm needs to be a bit smarter and also does checksum verification. If the sha is not there, the download needs to restart, if it is there, we need to compare the sha against the ISO and if they don't match re-trigger the download...

**Comment by budhrg** [comment]
> So our algorithm needs to be a bit smarter and also does checksum verification. If the sha is not there, the download needs to restart, if it is there, we need to compare the sha against the ISO and if they don't match re-trigger the download...

Agree :+1:

**Comment by hferentschik** [comment]
actually, we cannot purely rely on a checksum being in the cache, but that can be discussed when we implement this.

**Comment by hferentschik** [comment]
Resolved via #1124
